packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -170,6 +173,18 @@ class InnerBlocks extends Component {
 			/>
 		);
 
+		// Wrap context provider if (and only if) block has context to provide.
+		const blockType = getBlockType( block.name );
+		if ( blockType && blockType.providesContext ) {
+			const context = pick( block.attributes, blockType.providesContext );
```

Comment:
This might be something else to memoize if `useMemo` is used in `BlockContextProvider`.

Reply:
> This might be something else to memoize if `useMemo` is used in `BlockContextProvider`.

Memoized as of 9b96b6584.